api/src/services/graphql/resolvers/system-admin.ts

```diff
@@ -6,197 +6,207 @@ import { ExtensionsService } from '../../extensions.js';
 import { FieldsService } from '../../fields.js';
 import { RelationsService } from '../../relations.js';
 import { GraphQLService } from '../index.js';
-import type { BaseTypeComposers } from './system.js';
+import type { Schema } from '../schema/index.js';
+import { getFieldType } from './get-field-type.js';
+import { getRelationType } from './get-relation-type.js';
+import { getCollectionType } from './get-collection-type.js';
 
 export function resolveSystemAdmin(
 	gql: GraphQLService,
+	schema: Schema,
 	schemaComposer: SchemaComposer<GraphQLParams['contextValue']>,
-	{ Collection, Field, Relation, Extension }: BaseTypeComposers,
 ) {
-	if (gql.accountability?.admin === true) {
-		schemaComposer.Mutation.addFields({
-			create_collections_item: {
-				type: Collection,
-				args: {
-					data: toInputObjectType(Collection.clone('create_directus_collections'), {
-						postfix: '_input',
-					}).addFields({
-						fields: [
-							toInputObjectType(Field.clone('create_directus_collections_fields'), { postfix: '_input' }).NonNull,
-						],
-					}).NonNull,
-				},
-				resolve: async (_, args) => {
-					const collectionsService = new CollectionsService({
-						accountability: gql.accountability,
-						schema: gql.schema,
-					});
+	if (gql.accountability?.admin === false) {
```

Comment:
Account for the optional chaining which could return `undefined`

```suggestion
	if (!gql.accountability?.admin) {
```